### PR TITLE
operator: Sync OperatorStatus before and after invoking sync functions

### PR DESCRIPTION
**Pull request by yifan-gu**
- Import some cvo packages for the definition of the OperatorStatus and its clientset.
- Register the cvoclientset into the mco.
- Call syncOperatorStatus before and after invoking sync functions, also handle errors.

TODO: I'd like to separate the OperatorStatus clientset from the cvo clientset, maybe we should have the SyncOperatorStatus exported from there too as this would probably be used by all other operators as well? Thoughts?

Ref: https://jira.coreos.com/browse/CORS-771

### CODE REVIEW

**Comment by yifan-gu** [comment]
Also /cc @crawford . Do you think that it makes more sense to separate the `OperatorStatusesGetter` from the `ClusterversionV1Interface`?

File: pkg/operator/status.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// syncStatus applies the new condition to the mco's OperatorStatus object.
+func (optr *Operator) syncStatus(cond v1.OperatorStatusCondition) error {
+	if cond.Type == v1.OperatorStatusConditionTypeDegraded {
+		return fmt.Errorf("invalid cond %s", cond.Type)
+	}
+
+	// TODO(yifan): Fill in the Extention field for the status
```
Comment (abhinavdahiya) [inline pkg/operator/status.go]: The `Extension` field is only meant to provide extra details for the components.

File: Gopkg.lock [hunk 2]
```diff
@@ -2,62 +2,47 @@
 
 
 [[projects]]
```
Comment (abhinavdahiya) [inline Gopkg.lock]: Can you update your dep to latest version and rerun `./hack/update-vendor.sh`.
here's mine:
```console
installer git:(installconfig_in_old) ✗ dep version
dep:
 version     : v0.5.0
 build date  : 2018-07-26
 git hash    : 224a564
 go version  : go1.10.3
 go compiler : gc
 platform    : linux/amd64
 features    : ImportDuringSolve=false
```

**Pull request merged**